policies/container/required_labels.py

```
@@ -11,31 +18,59 @@ def main():
+                def_labels = definition.get_value_or_default(".labels", {})
+
+                # Labels from matching build commands
+                matching_build_labels = build_labels_by_dockerfile.get(path, {})
 
-        # Union of both sources
-        all_labels = {**dockerfile_labels, **build_labels}
+                # Union of both sources for this Dockerfile
+                all_labels = {**def_labels, **matching_build_labels}
 
-        missing = [l for l in required if l not in all_labels]
+                missing = [l for l in required if l not in all_labels]
```

Ruff (0.15.2)

[error] Ambiguous variable name: `l`

(E741)